### ISSUE TITLE
How to write backward function in RoundFunction?

### ISSUE THREAD

**Issue by SmallHedgehog** [sonack/How_to_write_cuda_extensions_in_pytorch#2]
I write backward function below, but I can't get gradients of input.

class RoundFunction(Function):
    ...
    def backward(self, grad_output):
        grad_in = torch.zeros_like(grad_output)
        round_backward_wrapper(grad_output, grad_in, grad_output.numel())
        return grad_in

roundFunc = RoundFunction()
x  = torch.randn(2, 2, requires_grad=True)
x = x.cuda()
y = roundFunc(x)
y.backward(torch.ones_like(y))
print(x.grad)  # get None

So how to solve it?

**Comment by sonack** [comment]
@SmallHedgehog  Hi, Hedgedog, because you write `x=x.cuda()`, then x became a non-leaf node which will not preserve grad, maybe you can refer to [this](https://discuss.pytorch.org/t/how-to-get-cuda-variable-gradient/1386).

**Comment by SmallHedgehog** [comment]
Solved, @sonack Thank you so much!
